File: android/java/app/src/main/java/com/daily/partyline/FooterAdapter.java
package com.daily.partyline;

import android.content.Intent;
import android.text.method.LinkMovementMethod;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.daily.partyline.databinding.ListItemFooterBinding;

public class FooterAdapter extends RecyclerView.Adapter<FooterAdapter.ViewHolder> {

    public String mShareLink;
    ListItemFooterBinding mBinding;

    public FooterAdapter(String shareLink) {
        mShareLink = shareLink;
    }

    public void setRoomUrl(String shareLink) {
        mShareLink = shareLink;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public FooterAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        mBinding = ListItemFooterBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(mBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull FooterAdapter.ViewHolder holder, int position) {
        holder.bind(mShareLink);
    }

    @Override
    public int getItemCount() {
        return 1;
    }

    @Override
    public int getItemViewType(final int position) {
        return R.layout.list_item_footer;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        ListItemFooterBinding mBinding;

        public ViewHolder(@NonNull ListItemFooterBinding binding) {
            super(binding.getRoot());
            mBinding = binding;
        }

        public void bind(String shareLink) {
            mBinding.textLink.setMovementMethod(LinkMovementMethod.getInstance());
            mBinding.shareButton.setOnClickListener((sender) -> {
                String roomCode = shareLink;

                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT, roomCode);
                sendIntent.setType("text/plain");

                Intent shareIntent = Intent.createChooser(sendIntent, null);
                mBinding.getRoot().getContext().startActivity(shareIntent);
            });
        }
    }
}

File: android/kotlin/app/src/main/java/com/daily/partyline/FooterAdapter.kt
package com.daily.partyline

import android.content.Intent
import android.text.method.LinkMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.daily.partyline.databinding.ListItemFooterBinding

class FooterAdapter(var shareLink: String?) : RecyclerView.Adapter<FooterAdapter.ViewHolder?>() {

    fun setRoomUrl(shareLink: String?) {
        this.shareLink = shareLink
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ListItemFooterBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(shareLink)
    }

    override fun getItemCount(): Int {
        return 1
    }

    override fun getItemViewType(position: Int): Int {
        return R.layout.list_item_footer
    }

    class ViewHolder(var binding: ListItemFooterBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(shareLink: String?) {
            binding.textLink.movementMethod = LinkMovementMethod.getInstance()
            binding.shareButton.setOnClickListener { sender: View? ->
                val sendIntent = Intent()
                sendIntent.action = Intent.ACTION_SEND
                sendIntent.putExtra(Intent.EXTRA_TEXT, shareLink)
                sendIntent.type = "text/plain"
                val shareIntent = Intent.createChooser(sendIntent, null)
                binding.getRoot().context.startActivity(shareIntent)
            }
        }

    }
}